apps/ios/GuideDogs/Code/Data/Models/Extensions/OSM Entity/GDASpatialDataResultEntity+Typeable.swift

```diff
@@ -12,22 +12,113 @@ extension GDASpatialDataResultEntity: Typeable {
     
     func isOfType(_ type: PrimaryType) -> Bool {
         switch type {
-        case .transit: return isOfType(.transitStop)
+        case .transit:
+            return isOfType(.transitStop)
+        case .food:
+            return isFood()
+        case .landmarks:
+            return isLandmarks()
+        case .park:
+            return isPark()
+        case .hotel:
+            return isHotel()
         }
     }
     
     func isOfType(_ type: SecondaryType) -> Bool {
         switch type {
-        case .transitStop: return isTransitStop()
+        case .transitStop:
+            return isTransitStop()
+        case .food:
+            return isFood()
+        case .landmarks:
+            return isLandmarks()
+        case .park:
+            return isPark()
+        case .hotel:
+            return isHotel()
         }
     }
     
     private func isTransitStop() -> Bool {
+        print("Raw superCategory: \(superCategory)")
         guard let category = SuperCategory(rawValue: superCategory) else {
             return false
         }
+        print("Mapped category: \(category)")
+        let isTransitLocation = category == .mobility && localizedName.lowercased().contains(GDLocalizedString("osm.tag.bus_stop").lowercased())
+
+        return isTransitLocation
+    }
+
+    private func isFood() -> Bool {
+        
+        guard let category = SuperCategory(rawValue: superCategory) else {
+            return false
+        }
+        print("Mapped category: \(category)")
+        
+        let restaurantTags = [
+            GDLocalizedString("osm.tag.restaurant"),
+            GDLocalizedString("osm.tag.fast_food"),
+            GDLocalizedString("osm.tag.cafe"),
+            GDLocalizedString("osm.tag.bar"),
+            GDLocalizedString("osm.tag.ice_cream"),
+            GDLocalizedString("osm.tag.pub"),
+            GDLocalizedString("osm.tag.coffee_shop")
```

Comment:
You don't need to use localized strings here, because the OSM tag will always be e.g. "restaurant" regardless of what language the name of the restaurant is in. I think you're handling this properly in landmarks and parks below.